### ISSUE TITLE
If a user account fails a defined number of attempts to idle with a defined time period - that account cannot idle until the idler is restarted

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#2418]
Started seeing this with a user account that was used for manual testing on Feb 28/March 1. At some point, its attempts to idle Jenkins reached a threshold, at which point, attempts to idle Jenkins for this account stopped. 

Can we:
* Get some definition on the number of attempts that must fail with what time period?
* Generate logging messages (info level) to assist the user and RH support staff to debug when/why idling is disabled
* Provide a per-account means to re-enable idling?

Thx!

**Comment by aslakknutsen** [comment]
@hferentschik doc a 'alg' doc somewhere?

**Comment by chmouel** [comment]
Just to answer the question about the rule set, we have this as default : 

https://github.com/fabric8-services/fabric8-jenkins-idler/blob/559b76b/openshift/jenkins-idler.app.yaml#L32-L39

As what each variable does is described in the comments here : 

https://github.com/fabric8-services/fabric8-jenkins-idler/blob/559b76b/internal/configuration/env_config.go#L72-L106

In plain english for the default configuration (may change in production) in our case it means : 

* Idler would detect by checking every `15 minutes`  if there is no activities happening on jenkins for the last `45 minutes` and if it that jenkins has no activities for that period it will attempt to idle it.
* It will try `3 times` to idle jenkins with an interval of `15 minutes` between each tries. 
* It would then wait for `30 minutes` before resetting that counter and start again,

I believe most of the idler functions are properly tracked under `debug level`. I am not entirely sure if chaning this would cause a too many log messages to show up. The `/api/idler/info/$namespace` call should give all the information needed about what's happening currently for that user, for example : 

```json
{
  "Name": "cboudjna",
  "ID": "fe823100-bb07-4ba6-a4a8-e2baf2ec3c6d",
  "ActiveBuild": {
    "metadata": {
      "name": "space3-1",
      "namespace": "cboudjna",
      "annotations": {
        "openshift.io/build.number": "1",
        "openshift.io/jenkins-namespace": "cboudjna-jenkins"
      },
      "Generation": 0
    },
    "status": {
      "phase": "Running",
      "startTimestamp": "2018-03-20T16:11:56Z",
      "completionTimestamp": "2018-03-21T07:40:42.455699Z"
    },
    "spec": {
      "replicas": 0,
      "Strategy": {
        "Type": "JenkinsPipeline"
      }
    }
  },
  "DoneBuild": {
    "metadata": {
      "annotations": {},
      "Generation": 0
    },
    "status": {
      "phase": "",
      "startTimestamp": "0001-01-01T00:00:00Z",
      "completionTimestamp": "0001-01-01T00:00:00Z"
    },
    "spec": {
      "replicas": 0,
      "Strategy": {
        "Type": ""
      }
    }
  },
  "JenkinsLastUpdate": "2018-03-21T07:42:32Z"
}
```

Having said that I think it may be a good idea to store there the reason why it failed to idled, I have added an issue here https://github.com/fabric8-services/fabric8-jenkins-idler/issues/171 to track this down

Hope this helps,

**Comment by bmicklea** [comment]
@ldimaggi @chmouel @qodfathr @aslakknutsen 

If I understand this issue correctly the impact would be that some number of OSIO users could get into a state where their Jenkins is always on. This means that (for them) it all works correctly but we (as the hoster of the service) are paying for Jenkins instances that may not be used but which are still active.

If that's accurate then I think we should remove the `P0` label as it has not negative effect on user experience only on our bank account.

**Comment by ldimaggi** [comment]
I think that that is correct - the net result is that Jenkins will always be "on" for these users.

**Comment by maxandersen** [comment]
+1 on removing P0 here; still need to be fixed but not a user visible problem and if we see this occuring too much we can restart idler.

@kbsingh agree ?

**Comment by containscafeine** [comment]
FYI: The idler issue which @chmouel opened was fixed - https://github.com/fabric8-services/fabric8-jenkins-idler/issues/171

**Comment by piyush1594** [comment]
@sthaha I think this is still the issue not sure. Do you have any inputs on this?

**Comment by piyush1594** [comment]
This needs to be resolved @sthaha
